aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -1497,6 +1498,137 @@ void fake_quant_grad_per_channel_cpu(TensorIterator &iter, int64_t quant_min, in
     });
 }
 
+template <typename T>
+void LayerNormKernelQuantizedImplInternal(
+    const Tensor& X,
+    const Tensor& gamma,
+    const Tensor& beta,
+    int64_t M,
+    int64_t N,
+    float eps,
+    Tensor* Y) {
+
+  using qVec = vec256::Vec256<T>;
+  using fVec = vec256::Vec256<float>;
+
+  DCHECK_EQ(X.numel(), M * N);
+  DCHECK(!gamma.defined() || gamma.numel() == N);
+  DCHECK(!beta.defined() || beta.numel() == N);
```

Comment:
Can we make these `TORCH_INTERNAL_ASSERT` unless we see a performance issue? That way we'll get better error messages and reports from users if there's a bug. Do we expect this kernel to be called in a tight loop?